### ISSUE TITLE
Any interest in providing an option to switch server from the official mobile apps and browser extensions?

### ISSUE THREAD

**Issue by GasimGasimzada** [padloc/padloc#746]
It would be super useful to self host the data but utilize the official apps and extensions to interact with our password manager since building custom versions of apps adds too much friction for self hosted solutions, especially when iOS app is needed since we need to either publish a custom version of app or sideload it, constantly certify etc + need to have a macOS device to do it.

I was thinking that what if the official extensions and apps had a way to set a server URL during log in. Would this something that is of interest to you?

**Comment by ReessKennedy** [comment]
+1 ... agreed. Bitwarden allows this.

**Comment by seaglade** [comment]
At the very least there should be some documentation on how to build them yourself for your own server. As it stands, the admin page isn't available as a docker container (and inexplicably is not part of the server) and the various apps as well as the extension are 100% undocumented. The maintainers really don't seem to understand that advertising your project as "able to be self hosted" and publishing a massive monorepo doesn't actually do much in the way of helping people host their own servers.

**Comment by giftedl** [comment]
> At the very least there should be some documentation on how to build them yourself for your own server. As it stands, the admin page isn't available as a docker container (and inexplicably is not part of the server) and the various apps as well as the extension are 100% undocumented. The maintainers really don't seem to understand that advertising your project as "able to be self hosted" and publishing a massive monorepo doesn't actually do much in the way of helping people host their own servers.

That was the plan,
> In-depth guides on how to host your own "productive" version of Padloc and how to build and distribute your own versions of the desktop and mobile apps are coming soon!

but that hasn't happened yet. Along with this, there is a difference between "able to be self-hosted" (like you said) and "open-source". The only one present in this project is the open-source part. No maintainer has ever said Padloc is an angel to self-host.

**Comment by working-name** [comment]
> No maintainer has ever said Padloc is an angel to self-host.

No, unfortunately it's almost impossible for the average fiddler/self-hoster. I ended up with local code modifications which I have to re-apply every pull... adds quite the maintenance burden and the possibility of breaking changes.

I've been watching the project for quite a while and I keep toying with it periodically. Still not able (#737) to get it to a point where I have it running to try an import. There's always an issue when I try to run it, and it gets down to modifying existing code vs just not having a good understanding of how it's supposed to be deployed. I believe that is because this codebase is not quite what you use internally, or at least not everything is checked into the repo (like setup/deployment scripts). Not having an admin interface is just not credible. Code isn't published yet is more likely. There's no way enterprise level clients would invest in a password manager without admin controls and auditing trails.

I've expressed my doubts in 2022 - about the financial feasibility of allowing for an easy self-hosted version - and I got a wrist slap, saying that the point of padloc is to be the easiest password manager to self host https://github.com/padloc/padloc/issues/605#issuecomment-1272334830. Period. That sounds great, but after 2+ years I cannot see much evidence of inching towards that noble goal. That makes me sad, but at the same time I hope enterprise money is flowing in, otherwise there won't be any padloc in the future.

<hr/>

I also find it pretty funny to think of your new dev onboarding session taking some 4+ hours trying to figure out how to get this running locally as a new install...😆 "Yeah, so just a quick `git clone` and `npm i` and you're good. Hmm ... likely not an npm error... what?! Oh wait, I think it was node 16. 14? Ok, so let's try 16 and see."

**Comment by giftedl** [comment]
> > No maintainer has ever said Padloc is an angel to self-host.
> 
> No, unfortunately it's almost impossible for the average fiddler/self-hoster. I ended up with local code modifications which I have to re-apply every pull... adds quite the maintenance burden and the possibility of breaking changes.
> 
> I've been watching the project for quite a while and I keep toying with it periodically. Still not able (#737) to get it to a point where I have it running to try an import. There's always an issue when I try to run it, and it gets down to modifying existing code vs just not having a good understanding of how it's supposed to be deployed. I believe that is because this codebase is not quite what you use internally, or at least not everything is checked into the repo (like setup/deployment scripts). Not having an admin interface is just not credible. Code isn't published yet is more likely. There's no way enterprise level clients would invest in a password manager without admin controls and auditing trails.
> 
> I've expressed my doubts in 2022 - about the financial feasibility of allowing for an easy self-hosted version - and I got a wrist slap, saying that the point of padloc is to be the easiest password manager to self host https://github.com/padloc/padloc/issues/605#issuecomment-1272334830. Period. That sounds great, but after 2+ years I cannot see much evidence of inching towards that noble goal. That makes me sad, but at the same time I hope enterprise money is flowing in, otherwise there won't be any padloc in the future.
> 
> <hr/>
> 
> I also find it pretty funny to think of your new dev onboarding session taking some 4+ hours trying to figure out how to get this running locally as a new install...😆 "Yeah, so just a quick `git clone` and `npm i` and you're good. Hmm ... likely not an npm error... what?! Oh wait, I think it was node 16. 14? Ok, so let's try 16 and see."

I will say something; I am somewhat familiar with this projects code and if anyone wants to start a fork with me (I'd rather not go solo), I could make it more friendly to self-host and we could collectively maintain a new version. Hit me up on Discord if you'd like.
